sdk/dotnet/Pulumi/Testing/IMocks.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Pulumi.Testing
+{
+    /// <summary>
+    /// Hooks to mock the engine that provide test doubles for offline unit testing of stacks.
+    /// </summary>
+    public interface IMocks
```

Comment:
don't like the name.  but not sure what a better name would be

Reply:
Its sole usage is being an argument of `TestAsync` method, so mocks is what it is. We can go for `IEngine`, `IRemoteCalls`, ... and keep `mocks` for the parameter name

Reply:
This aligns with the naming in TS/Python, so I'll leave it at `IMocks`.